.github/BUILD.md

```diff
@@ -13,81 +13,109 @@ If the answer is yes, please read on. Otherwise, make sure to head on over [to t
--   [Node.js v20+](https://nodejs.org/en/download/)
--   npm v10+ (installed with Node.js)
--   [Git command line](https://git-scm.com/download/)
+Whilst most of the backoffice code lives in `Umbraco.Web.UI.Client`, the login screen is in a separate project. If you do any work with that you can build with:
 
-The easiest way to get started is to open `umbraco.sln` in Visual Studio.
+```
+cd <solution root>\src\Umbraco.Web.UI.Login
+npm run build
+```
 
-Umbraco is a C# based codebase, which is mostly ASP.NET Core MVC based. You can make changes, build them in Visual Studio, and hit <kbd>F5</kbd> to see the result. There are two web projects in the solution, `Umbraco.Web.UI.Client` and `Umbraco.Web.UI.Login`. They each have their own build process and can be run separately. The `Umbraco.Web.UI` project is the main project that hosts the back office and login screen. This is the project you will want to run to see your changes. It will automatically restore and build the client projects when you run it.
+### Full-stack changes
 
-If you want to watch the UI Client to `Umbraco.Web.UI` then you can open a terminal in `src/Umbraco.Web.UI.Client` where you can run `npm run dev:mock` manually. This will launch the Vite dev server on http://localhost:5173 and watch for changes with mocked API responses.
+If working across both front and back-end, follow both methods and use `dotnet watch`, or re-run `dotnet run` (or `dotnet build` followed by `dotnet run --no-build`) whenever you need to update the back-end code.
 
-You can also run `npm run dev:server` to run the Vite server against a local Umbraco instance. In this case, you need to have the .NET project running and accept connections from the Vite server. Please see the Umbraco.Web.UI.Client README.md [Run against a local Umbraco instance](../src/Umbraco.Web.UI.Client/.github/README.md#run-against-a-local-umbraco-instance) for more information.
+Request and response models used by the management APIs are made available client-side as generated code. If you make changes to the management API, you can re-generate the typed client code with:
 
-**The initial C# build might take a _really_ long time (seriously, go and make a cup of coffee!) - but don't worry, this will be faster on subsequent runs.**
+```
+cd <solution root>\src\Umbraco.Web.UI.Client
+npm run generate:server-api-dev
+```
 
-When the page eventually loads in your web browser, you can follow the installer to set up a database for debugging. You may also wish to install a [starter kit][starter kits] to ease your debugging.
+Please also update the `OpenApi.json` file held in the solution by copying and pasting the output from `/umbraco/swagger/management/swagger.json`. Be careful to undo any automated formatting carried out after pasting by the IDE.
```

Comment:
What happens when I let my IDE autoformat? Can we prevent this?

Reply:
It doesn't do any harm... it's just that your check-in will show a lot more changes than you've actually made.  It's mostly around collections - the default output from Swagger is `[ ]` but Visual Studio prefers them as `[]`.  So if you let it auto-format you get a lot more lines changed in your PR.

Reply:
Ah okay! I am not too worried about this, putting it in here makes me think something will break. Easy for reviewers to tap the "Hide whitespace" checkbox. 

![image](https://github.com/user-attachments/assets/bf74282f-3d55-4b4f-9f4f-b9c9eb80fea7)

My suggestion would be to leave this hint out. 

```suggestion
Please also update the `OpenApi.json` file held in the solution by copying and pasting the output from `/umbraco/swagger/management/swagger.json`.
```